pkg/restful/resource.go

```diff
@@ -183,6 +189,98 @@ func (ar *AbstractResource) GetRouter() chi.Router {
 	return ar.router
 }
 
+func (ar *AbstractResource) parseURLParamValue(paramValue string) interface{} {
+	parsedBool, err := strconv.ParseBool(paramValue)
+	if err == nil {
+		return parsedBool
+	}
+
+	parsedInt, err := strconv.ParseInt(paramValue, 10, 64)
+	if err == nil {
+		return parsedInt
+	}
+
+	parsedUint, err := strconv.ParseUint(paramValue, 10, 64)
+	if err == nil {
+		return parsedUint
+	}
+
+	parsedFloat, err := strconv.ParseFloat(paramValue, 10)
+	if err == nil {
+		return parsedFloat
+	}
+
+	return paramValue
+}
+
+func (ar *AbstractResource) GetURLParams(paramKey string, request *http.Request) []interface{} {
+	paramValues, ok := request.URL.Query()[paramKey]
+	if !ok || len(paramValues) == 0 {
+		return nil
+	}
+
+	var values []interface{}
+	for _, value := range paramValues {
+		values = append(values, ar.parseURLParamValue(value))
+	}
+
+	return values
+}
+
+func (ar *AbstractResource) GetURLParam(paramKey string, request *http.Request) interface{} {
+	paramValues, ok := request.URL.Query()[paramKey]
+	if !ok || len(paramValues) == 0 {
+		return nil
+	}
+
+	return ar.parseURLParamValue(paramValues[0])
+}
+
+func (ar *AbstractResource) GetBoolURLParam(paramKey string, request *http.Request) bool {
```

Comment:
for all below functions including this one, returning a default should be explicit by user - such as
`GetURLParamBoolOrDefault` and let user provide a default